google/cloud/storage/tests/run_integration_tests_production.sh

```diff
@@ -23,6 +23,14 @@ echo
 echo "Running storage::internal::CurlRequest integration test."
 ./curl_request_integration_test
 
+#echo
```

Comment:
Add a TODO for what needs to be done before we can uncomment these?

Reply:
Actually I do not know if they ever should be running in the "production" integration tests. They would be hitting `https://nghttp2.org/httpbin` which seems wrong/abusive.  And we basically run the same thing in the `run_integration_tests_testbench.sh`.  I am inclined to remove all these curl tests from the `_production.sh` script, created #922 for that.